controllers/vspherevm_controller.go

```diff
@@ -607,7 +607,7 @@ func (r vmReconciler) fetchClusterModuleInfo(clusterModInput fetchClusterModuleI
 	logger := r.Logger.WithName(machine.Namespace).WithName(machine.Name)
 
 	input := util.FetchObjectInput{
```

Comment:
We need to refactor this structure and the related functions  in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/pkg/util/fetch_object.go, to remove the `context.Context` which is our final goal, if there's not too complicated in this PR. Otherwise, please leave a TODO in the comment in #2295 
```
type FetchObjectInput struct {
	context.Context //nolint:containedctx
	ctrlclient.Client
	Object ctrlclient.Object
}
```

Reply:
This is not complicated, so I remove context from the `FetchObjectInput struct.